src/server/game/AntiCheat/AntiCheatMgr.cpp

```diff
@@ -0,0 +1,523 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+Nemesis Anti Cheat
+Inspired by the old passive anti cheat (https://github.com/TrinityCore/TrinityCoreCustomChanges/tree/3.3.5-passive_anticheat)
+
+Authors:
+https://github.com/sveN295
+https://github.com/Jinnaix
+With feedback and support from https://github.com/acidmanifesto
+*/
+
+#include "AntiCheatMgr.h"
+#include "DatabaseEnv.h"
+#include "Map.h"
+#include "DBCStores.h"
+#include "Realm.h"
+#include "Log.h"
+
+AntiCheatMgr::AntiCheatMgr(Player* player)
+{
+    this->player = player;
```

Comment:
if possible avoid using this-> , for example renaming the class variable to _player